packages/generator/src/edmx-to-vdm/v4/function-import.ts

```diff
@@ -34,34 +35,40 @@ export function generateFunctionImportsV4(
   const functions = parseFunctions(serviceMetadata.edmx.root);
   const functionImports = parseFunctionImports(serviceMetadata.edmx.root);
 
-  return functionImports.map(functionImport => {
-    const edmxFunction = findFunctionForFunctionImport(
-      functions,
-      functionImport
-    );
+  return functionImports
+    .map(functionImport => {
+      const edmxFunction = findFunctionForFunctionImport(
+        functions,
+        functionImport
+      );
+      // TODO 1571 remove when supporting entity type as parameter
+      if (!edmxFunction || hasUnsupportedParameterTypes(edmxFunction)) {
+        return undefined;
+      }
```

Comment:
Sorry @jjtang1985, what I forgot in our discussion last week was, that the error handling (logging) is still in the find method which is the wrong place in my opinion and that this is here to stay (unlike the `hasUnsupportedParameterTypes` check only).

Consider the following:
```ts
type JoinedFunctionImportData = { functionImport: EdmxFunctionImport, function: EdmxFunction };

function joinFunctionImportData(functionImports: EdmxFunctionImport[], functions: EdmxFunction[]): JoinedFunctionImportData[] {
  const functionImportsWithoutFunctions = [];
  const joinedFunctionImportData = functionImports.reduce((joined, functionImport) => {
    const edmxFunction = findFunctionForFunctionImport(
      functions,
      functionImport
    );
    
    if(edmxFunction) {
      return [
        ...joined, {
          functionImport,
          function: edmxFunction
        }
      ];
    }
    functionImportsWithoutFunctions.push(functionImport.Name);
    return joined;
  }, []);

  if (functionImportsWithoutFunctions) {
    logger.warn(functionImportsWithoutFunctions);
  }
  return joinedFunctionImportData;
}

export function generateFunctionImportsV4(
  serviceMetadata: ServiceMetadata,
  entities: VdmEntity[],
  complexTypes: Omit<VdmComplexType, 'factoryName'>[],
  formatter: ServiceNameFormatter
): VdmFunctionImport[] {
  const functions = parseFunctions(serviceMetadata.edmx.root);
  const functionImports = parseFunctionImports(serviceMetadata.edmx.root);
  const joinedFunctionData = joinFunctionImportData(functionImports, functions);

  return joinedFunctionData
    // TODO 1571 remove when supporting entity type as parameter
    .filter(({ function: edmxFunction}) => !hasUnsupportedParameterTypes(edmxFunction))
    .map(({ functionImport, function: edmxFunction }) => {
    const httpMethod = 'get';
    const swaggerDefinition = swaggerDefinitionForFunctionImport(
      functionImport.Name,
      httpMethod,
      serviceMetadata.swagger
    );

    return {
      ...transformFunctionImportBase(
        functionImport,
        edmxFunction.Parameter,
        swaggerDefinition,
        formatter
      ),
      httpMethod,
      returnType: parseFunctionImportReturnTypes(
        edmxFunction.ReturnType?.Type,
        entities,
        complexTypes,
        extractResponse,
        serviceMetadata.edmx.oDataVersion
      )
    };
  });
}
```